datalad/interface/ls.py

```diff
@@ -193,6 +205,88 @@ def annex_local_size(self):
+
+    __slots__ = ['_path', '_info', 'repo']
+
+    def __init__(self, path, repo=None):
+        self._path = path  # fs path to the node, can be overridden
+        self._info = None
+        self.repo = DsModel(Dataset(path)) if self.type_ == 'annex' else repo  # parent repository associated with node
+        self._branch = None
+
+    @property
+    def symlink(self):
+        """if symlink returns path the symlink points to else returns None"""
+        if islink(self._path):                    # if symlink
+            target_path = readlink(self._path)    # find link target
+            # convert to absolute path if not
+            target_path = opj(dirname(self._path), target_path) if not isabs(target_path) else target_path
+            return target_path if exists(target_path) else None
+        return None
+
+    @property
+    def date(self):
+        """Date of last modification"""
+        if self.type_ is not ['git', 'annex']:
+            return lstat(self._path).st_mtime
+        else:
+            super(self.__class__, self).date
+
+    @property
+    def size(self):
+        """Size of the node computed based on its type"""
+        type_ = self.type_
+        if not type_:
+            return -1
+        if 'annex' == type_:
+            if self.annex_local_size:
+                return self.annex_local_size
+            else:
+                return Runner().run(['du', '-s', self._path])[0].split('\t')[0]
+        elif 'git' == type_:
+            return self.git_local_size
+        elif 'file' == type_:
+            return lstat(self._path).st_size
+        elif type_ in ['link', 'link-broken']:
+            if self.repo.repo.is_under_annex(self._path):
+                return self.repo.repo.info(self._path, batch=True)['size']
+            return 0 if type_ == 'broken-link' else lstat(self.symlink).st_size
+        elif 'dir' == type_:
+            return Runner().run(['du', '-s', self._path])[0].split('\t')[0]  # lstat(self._path).st_size
+        else:
+            return -1
+
+    @property
+    def type_(self):
+        """outputs the node type
+
+        Types: link, link-broken, file, dir, annex-repo, git-repo"""
+        if islink(self.path):
+            return 'link' if self.symlink else 'link-broken'
+        elif isfile(self.path):
+            return 'file'
+        elif exists(opj(self.path, ".git", "annex")):
+            return 'annex'
+        elif exists(opj(self.path, ".git")):
+            return 'git'
+        elif isdir(self.path):
+            return 'dir'
+        else:
+            return None
+
+    @property
+    def git_local_size(self):
+        """computes the disk space used by unpacked object files in the git repository"""
+        try:
+            describe, outerr = self.repo._git_custom_command([], ['git', 'count-objects', '-v'])[0].split('\n')
+            size = [item for item in describe if 'size: ' in item][0].split(': ')
+            return int(size[1])
+        except:
```

Comment:
catch actual exception and use `exc_str`
but first check why it fails at all, try to make it conditioned without invoking git without necessity